src/inference/kernel_dsl.jl

```diff
@@ -13,6 +13,11 @@ end
 is_custom_primitive_kernel(::Function) = false
 check_is_kernel(::Function) = false
 
+_is_custom_primitive_kernel = GlobalRef(Gen, :is_custom_primitive_kernel)
+_check_is_kernel = GlobalRef(Gen, :check_is_kernel)
+_check_observations = GlobalRef(Gen, :check_observations)
+_reversal = GlobalRef(Gen, :reversal)
```

Comment:
If we're doing these as globals, we should probably make them `const` -- it won't affect run-time behavior, but it will probably make the macros run just a little bit faster. Fine either way though!